lib/pharos/addon.rb

```diff
@@ -188,6 +188,10 @@ def apply_uninstall
       end
     end
 
+    def notify(msg)
```

Comment:
Not sure if `notify_user` would be better name

Reply:
```ruby
def post_install_message(msg)
  @cluster_context['post_install_messages'][self.class.addon_name] ||= []
  @cluster_context['post_install_messages'][self.class.addon_name].concat(msg.lines)
end
```

with this single-string approach you need to do something like:
```
notify "foo\nfoofoo\nfoo\n"
```

and if you want to add to it:

```
notify "foofoo\nfoofoo"
@post_install_message << "\nfoofoo"
```

Reply:
oh, addons don't have direct access to cluster context. maybe they should.

Reply:
We were discussing about cluster_context in addons and decided not to pass it there.. is it really a good practice to "leak" cluster_context to addons?

Reply:
It already leaks the stuff that is updated directly to `config.hosts` by `gather_facts` et al, like  `host.checks` and whatever.

Maybe `cluster_context` could actually be an instance variable of `config` anyways. then addons, phases and host_configurers could all reach it through `config.cluster_context` since they all get the config.

Reply:
btw i don't know how threadsafe the cluster_context thing is anyways.. maybe it should be an instance of `Concurrent::Hash` instead of a regular hash (dry already brings in `concurrent-ruby`).